### DISCUSSION TITLE
Help with migrating to 1.0

### DISCUSSION THREAD

**Discussion by TomasMikula** [planet42/Laika#552] [Q&A] [answered]
Hi, thanks for keeping Laika great!

I'd like to migrate [my Markdown-to-HTML transformer](https://github.com/TomasMikula/libretto/blob/main/project/LaikaMarkdownToHtml.scala) from 0.19.5 to 1.0.0 and hit a couple of obstacles:
 - What's the replacement for `TransformerBuilder#withTheme()`?
 - How to run the transformer on source directory and target directory?

Thanks for your help!

**Comment by jenshalm** [comment]
I'm a little unclear on what the actual problem is, as the two things you mention haven't really changed.

Can you post a code snippet that used to work in 0.19.5 and doesn't in 1.0.0?

**Comment by jenshalm** [comment]
Actually, even without seeing code, I suspect it's most likely just imports (there was a fairly big repackaging for 1.0).

For your case the issue might be that `laika.io.implicits._` became `laika.io.syntax._`

**Comment by TomasMikula** [comment]
Thanks for the reply.

The code snippet that used to work in 0.19.5 is under the link above, but here it is for completeness:

```scala
import cats.effect.IO
import cats.effect.unsafe.implicits.global
import laika.api.Transformer
import laika.format.{HTML, Markdown}
import laika.helium.Helium
import laika.helium.config.TextLink
import laika.io.implicits._
import laika.markdown.github.GitHubFlavor
import laika.parse.code.SyntaxHighlighting
import laika.theme.Theme

Transformer
  .from(Markdown)
  .to(HTML)
  .withRawContent
  .using(GitHubFlavor, SyntaxHighlighting)
  .parallel[IO]
  .withTheme(
    Helium.defaults
      .site.topNavigationBar(
        homeLink = TextLink.external("https://github.com/TomasMikula/libretto", "GitHub"),
      )
      .site.fontResources(/* Empty overrides Default! We use our custom CSS for the fonts. */)
      .build
  )
  .build
  .use { transformer =>
    transformer
      .fromDirectory(srcDir)
      .toDirectory(tgtDir)
      .transform
  }
  .unsafeRunSync()
```

I have managed to find `GitHubFlavor` and `SyntaxHighlighting`, removed `.parallel[IO]`, and found that `.build` no longer returns a resource.

But the `.withTheme` and `.fromDirectory`, `.toDirectory` methods appear non-existent.

**Comment by jenshalm** [comment]
Yes, as I suspected, my previous answer applies to your code.

**Reply by TomasMikula** [reply]
I am getting

```
[info] compiling 1 Scala source to /Users/tomas/projects/libretto/project/target/scala-2.12/sbt-1.0/classes ...
[error] /Users/tomas/projects/libretto/project/LaikaMarkdownToHtml.scala:21:8: value withTheme is not a member of laika.api.builder.TransformerBuilder[laika.api.format.TagFormatter]
[error] possible cause: maybe a semicolon is missing before `value withTheme'?
[error]       .withTheme(
[error]        ^
[error] one error found
```

even after importing `laika.io.syntax._`

**Reply by jenshalm** [reply, marked as answer]
This error implies that the code you are trying to compile is not exactly the code you posted. I assume you removed the call to `.parallel[IO]` which is still needed to get into the richer builder API. Without it you are building a base transformer that can only transform from string to string in memory and does not support themes.

**Reply by TomasMikula** [reply]
Ah, you are right, I removed it because it did not compile 😄 🤦‍♂️ 
Thanks for your help!
